### PR TITLE
Rework chbench snapshot generation

### PR DESCRIPTION
**Pull request by cirego**
This PR introduces a number of changes around how we generate snapshots from chbench, for use with the ingest tests. First, our snapshot generation code was never 100% complete, as it never wrote out the `source_offsets` (I had created it by hand and it was part of the S3 archive). Snapshot_view_states now captures the offsets for each source.

Second, `offsets.json` hardcoded the view_name to topic name. Instead of duplicating this information, add code to Peeker to archive the configuration used to generate the snapshot. Included in this configuration file are the sources that comprise each view (currently expected to be a 1 to 1 mapping for our ingest benchmarks).

Third, the TOML config file for Peeker is interpreted by Peeker (it substitutes environment variables) and cannot be directly loaded by other programming languages. To fix this, I added a method to Peeker to write out the computed version of the configuration (and loaded it into TOML to validate the config).

Fourth, the TOML config file for Peeker could not be serialized due to `sleep_ms` being a Duration field and not being the last field in the struct. I moved this field to the end of each struct where mentioned, but this created a second problem. Our deserializer converted a single string into a Duration object, which is serialized as a table, thereby breaking symmetry in serialize / deserialize. I removed the custom deserialization so that we can load the serialized file using the same deserialization logic.

I think we can write a SQL query / some recursive logic to programmatically get the source names for each view, but this current PR works for saving snapshots and archiving the Peeker config file.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5330)

### CODE REVIEW

**Comment by ruchirK** [comment]
not sure if you're aware but you can either run `bin/pre-push` or follow the instructions in `misc/githooks/pre-push` to make a pre push git hook that runs fmt checks and clippy before pushing. makes iterating on this a little nicer :)

**Comment by cirego** [comment]
> not sure if you're aware but you can either run `bin/pre-push` or follow the instructions in `misc/githooks/pre-push` to make a pre push git hook that runs fmt checks and clippy before pushing. makes iterating on this a little nicer :)

Thanks for the tip! I was aware of `bin/lint`, but not `bin/pre-push`.

**Review by sploiselle — APPROVED** [review]
I didn't write enough of this code to understand its shape, but LGTM

File: src/peeker/src/args.rs [hunk 1]
```diff
@@ -141,6 +152,27 @@ pub fn print_config_supplied(config: Config) {
     }
 }
 
+pub fn write_config_supplied(config_path: Option<&str>, outfile: &str) {
+    let config_contents = toml::to_string(&load_raw_config(config_path));
+    match &config_contents {
+        Ok(contents) => match fs::write(outfile, contents) {
```
[the remark this answers is not in the document]
Comment (cirego) [reply src/peeker/src/args.rs:158]: Thanks -- will do!

**Pull request merged**